src/cat/cat.rs

```diff
@@ -241,6 +248,8 @@ fn open(path: &str) -> CatResult<InputHandle> {
     if path == "-" {
         let stdin = stdin();
         return Ok(InputHandle {
+            #[cfg(any(target_os = "linux", target_os = "android"))]
+            file_descriptor: stdin.as_raw_fd(),
             reader: Box::new(stdin) as Box<Read>,
```

Comment:
I mentioned this on the other PR as well, but could you change this to avoid `Box`?